docs/download.md

```diff
@@ -23,32 +23,39 @@ license: |
+|:-----------:|:--------------------------------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------:|
+| Source Code | [src](https://downloads.apache.org/celeborn/celeborn-0.4.1/apache-celeborn-0.4.1-source.tgz) | [sha512](https://downloads.apache.org/celeborn/celeborn-0.4.1/apache-celeborn-0.4.1-source.tgz.sha512) | [asc](https://downloads.apache.org/celeborn/celeborn-0.4.1/apache-celeborn-0.4.1-source.tgz.asc) |
+|   Binary    |  [bin](https://downloads.apache.org/celeborn/celeborn-0.4.1/apache-celeborn-0.4.1-bin.tgz)   |  [sha512](https://downloads.apache.org/celeborn/celeborn-0.4.1/apache-celeborn-0.4.1-bin.tgz.sha512)   |  [asc](https://downloads.apache.org/celeborn/celeborn-0.4.1/apache-celeborn-0.4.1-bin.tgz.asc)   |
 
-|             |                                                           Download from ASF                                                           |                                                                  Checksum                                                                   |                                                                Signature                                                                |
-|:-----------:|:-------------------------------------------------------------------------------------------------------------------------------------:|:-------------------------------------------------------------------------------------------------------------------------------------------:|:---------------------------------------------------------------------------------------------------------------------------------------:|
-| Source Code | [src](https://www.apache.org/dyn/closer.lua/incubator/celeborn/celeborn-0.4.0-incubating/apache-celeborn-0.4.0-incubating-source.tgz) |   [sha512](https://downloads.apache.org/incubator/celeborn/celeborn-0.4.0-incubating/apache-celeborn-0.4.0-incubating-source.tgz.sha512)    |    [asc](https://downloads.apache.org/incubator/celeborn/celeborn-0.4.0-incubating/apache-celeborn-0.4.0-incubating-source.tgz.asc)     |
-| Binary      |  [bin](https://www.apache.org/dyn/closer.lua/incubator/celeborn/celeborn-0.4.0-incubating/apache-celeborn-0.4.0-incubating-bin.tgz)   |     [sha512](https://downloads.apache.org/incubator/celeborn/celeborn-0.4.0-incubating/apache-celeborn-0.4.0-incubating-bin.tgz.sha512)     |      [asc](https://downloads.apache.org/incubator/celeborn/celeborn-0.4.0-incubating/apache-celeborn-0.4.0-incubating-bin.tgz.asc)      |
+### 0.4.0-incubating (2024-02-06)
```

Comment:
remove it. we only need to reserve the latest patched version for each feature serial.